hips/hip-00xx-sequencing.md

```diff
@@ -0,0 +1,192 @@
+---
+hip: 9999
+title: "Better Support for Resource Creation Sequencing"
+authors: [ "Joe Beck <joebeck5705@gmail.com>", "Evans Mungai <mbuevans@gmail.com>" ]
+created: "2024-11-15"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+This HIP is to propose a new featureset in Helm 4 to provide Application developers, who create heam charts for their applications, a well supported way of defining what order chart resources and chart dependencies should be deployed to Kubernetes.
+
+## Motivation
+
+Today, to accomplish resource sequencing you have two options. The first is leveraging helm hooks, the second is building required sequencing into the application (via startup code or init containers). The existing hooks and weights can be tedious to build and maintain for Application developers, and built-in app sequencing can unecessarily increase complexity of a Helm application that needs to be maintained by Application Developers. Helm as a package manager should be capable of enabling developers to properly sequence how their applications are deployed, and by providing such a mechanism to developers, this will significantly improve the Application developer's experience.
+
+Additionally, Helm currently doesn't provide a way to sequence when chart dependencies are deployed, and this featureset would ideally address this.
```

Comment:
When it comes to sequencing, I'm not sure what you mean is clear in the document. I could come to two different conclusions:

1. The ordering of resources when they are uploaded to the k8s API server in one push
2. An ordered set of communications to the k8s API server where set it sent and completed prior to the next one being sent

Can you please provide some more clarity on what you mean.

Reply:
2 is what we're going for here. I'll make a note to try and make that more clear.

Reply:
Clarified what "resource sequencing" meant in the HIP